packages/itwin/quantity-formatting/src/components/quantityformat/internal/ScientificType.tsx

```diff
@@ -0,0 +1,94 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+
+import * as React from "react";
+import type { FormatProps } from "@itwin/core-quantity";
+import { parseScientificType, ScientificType as CoreScientificType } from "@itwin/core-quantity";
+import { IconButton, LabeledSelect } from "@itwin/itwinui-react";
+import type { SelectOption } from "@itwin/itwinui-react";
+import { SvgHelpCircularHollow } from "@itwin/itwinui-icons-react";
+import { useTranslation } from "../../../useTranslation.js";
+import "../FormatPanel.scss";
+
+/** Properties of [[ScientificType]] component.
+ * @internal
+ */
+export interface ScientificTypeProps {
+  formatProps: FormatProps;
+  onChange?: (format: FormatProps) => void;
+  disabled?: boolean;
+}
+
+/** Component to show/edit scientific type.
+ * @internal
+ */
+export function ScientificType(props: ScientificTypeProps) {
+  const { formatProps, onChange, disabled = false } = props;
+  const { translate } = useTranslation();
+
+  const handleSetFormatProps = React.useCallback(
```

Comment:
Looks like this function can be inlined in `handleScientificTypeChange`. This applies to quite a few files in where pattern like this is used:

```tsx
const handleSet* = React.useCallback((newProps) => {
  onChange(newProps);
}, [deps]);

const handle*Change = React.useCallack(() => {
  // handle change
  handleSet*(newProps);
}, [otherDeps, handleSet*]);
```

In my opinion it adds unecessary complexity when reading code and could be inlined instead:
```tsx
const handle*Change = React.useCallack(() => {
  // handle change
  onChange(newProps);
}, [otherDeps, handleSet*]);
```

I would suggest going through all components and checking for functions defined using `useCallback` that are used only in other single function defined with `useCallback`.

Reply:
thanks, made the change here and in other components